src/components/Form.vue

```diff
@@ -29,17 +31,15 @@ export default {
   },
   methods: {
     ...mapActions(["login"]),
-    loginClick() {
-      if (this.username !== "Abhay") {
-        return alert("Username is Abhay");
-      } else if (this.password !== "123") {
-        return alert("Password is 123");
-      } else {
-        this.login({
-          username: this.username,
-          password: this.password,
-        });
+    async loginClick() {
```

Comment:
you are not checking whether the values are coming in username & password or not,
if in case you have user HTML validation, protect you API call as well